adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -121,6 +125,102 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
     replaceRdd(tFn(rdd))
   }
 
+  private[rdd] def sorted: Boolean
+  private[rdd] def partitionMap: Option[Seq[Option[(ReferenceRegion, ReferenceRegion)]]]
+
+  /**
+   * This repartition method repartitions all data in this.rdd and distributes it as evenly as possible
+   * into the number of partitions provided. The data in this.rdd is already sorted (by ReferenceRegion.start)
+   * so we ensure that it is kept that way as we move data around.
+   *
+   * Data is repartitioned in lists to maintain the sorted order. After the repartition, the lists are placed
+   * in the correct order based on their original sorted order.
+   *
+   * @param partitions the number of partitions to repartition this.rdd into
+   * @return a new SortedGenomicRDDMixIn with the RDD partitioned evenly
+   */
+  private def evenlyRepartition(partitions: Int)(implicit tTag: ClassTag[T]): GenomicRDD[T, U] = {
+    require(sorted, "Cannot evenly repartition an unsorted RDD")
+    // the average number of records on each node will help us evenly repartition
+    val average: Double = rdd.count.toDouble / partitions
```

Comment:
+1
I would say to keep this approach in for now, but to also include a sampling approach. You may want to look at Spark's RangePartitioner.